components/QuizForum/index.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import Modal from './model';
+
+class QuizForum extends React.Component {
+  constructor(props) {
+    super(props);
+    this.props.checkLogin();
+    this.state = { isModalOpen: false, post: {} };
+  }
+  openModal() {
+    this.setState({ isModalOpen: true });
+  }
+
+  closeModal() {
+    this.setState({ isModalOpen: false });
+  }
+  handlePostChange = (e, field) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ ...this.state, post: { [field]: e.target.value } });
+  };
+  showLogInLink = () => {
+    const { user, currentQuestion, isAuthenticated } = this.props;
+    if (!isAuthenticated()) {
+      if (currentQuestion.stimulus) {
+        return (
+          <h4 style={{ textAlign: 'center' }}>
+            Please{' '}
+            <a style={{ cursor: 'pointer' }} onClick={this.props.login}>
+              Log In{' '}
+            </a>
+            to ask a question on the forum.
+          </h4>
+        );
+      }
+      return null;
+    }
+    return (
+      <div>
+        {currentQuestion.prompt && (
+          <button onClick={() => this.openModal()}>Open modal</button>
+        )}
+        <Modal
+          isOpen={this.state.isModalOpen}
+          onClose={() => this.closeModal()}
+        >
+          <form style={{ margin: 10 }}>
```

Comment:
since we have `redux-form` set up, it should be easy to refactor this to use [redux-forms](https://redux-form.com/7.1.1/docs/gettingstarted.md/#step-2-of-4-form-component)

Reply:
check